capture/capture.go

```diff
@@ -271,27 +253,39 @@ func (l *Listener) PcapHandle(ifi net.Interface) (handle *pcap.Handle, err error
 }
 
 // SocketHandle returns new unix ethernet handle associated with this listener settings
-func (l *Listener) SocketHandle(ifi net.Interface) (handle Socket, err error) {
-	handle, err = NewSocket(ifi)
-	if err != nil {
-		return nil, fmt.Errorf("sock raw error: %q, interface: %q", err, ifi.Name)
+func (l *Listener) socketHandle(ifi net.Interface) (handle *socket, err error) {
+	c := raw.DefaultConfig()
+	if l.BufferSize > 0 {
+		c.ReadBufferSize = int64(l.BufferSize)
 	}
-	if err = handle.SetPromiscuous(l.Promiscuous || l.Monitor); err != nil {
-		return nil, fmt.Errorf("promiscuous mode error: %q, interface: %q", err, ifi.Name)
+	if l.BufferTimeout > 0 {
+		if l.BufferTimeout < time.Millisecond {
+			l.BufferTimeout = time.Millisecond
+		}
 	}
+	c.ReadTimeout = l.BufferTimeout.Milliseconds()
+	c.Promiscuous = l.Promiscuous || l.Monitor
 	if l.BPFFilter != "" {
 		if l.BPFFilter[0] != '(' || l.BPFFilter[len(l.BPFFilter)-1] != ')' {
 			l.BPFFilter = "(" + l.BPFFilter + ")"
 		}
 	} else {
 		l.BPFFilter = l.Filter(ifi)
 	}
-	if err = handle.SetBPFFilter(l.BPFFilter); err != nil {
-		handle.Close()
+	snaplen := 128 * 1024
+	filter, err := pcap.CompileBPFFilter(layers.LinkTypeEthernet, snaplen, l.BPFFilter)
+	if err != nil {
 		return nil, fmt.Errorf("BPF filter error: %q%s, interface: %q", err, l.BPFFilter, ifi.Name)
 	}
-	handle.SetLoopbackIndex(int32(l.loopIndex))
-	return
+	f := make([]bpf.RawInstruction, len(filter))
+	for i := range filter {
```

Comment:
What this lines do?

Reply:
convert `[]pcap.BPFInstruction` to `[]bpf.RawInstruction`